### ISSUE TITLE
Transfer should be able to select all rows between Click and Shift + Click

### ISSUE THREAD

**Issue by anuragnandi** [ant-design/ant-design#33934]
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
I am using the Transfer component in many pages of my web application, Sometimes there is a list with 200  records, currently the only way to do it is Select all, Select one by one, Inverse select which are all not viable, It would be nice if there was a way we can drag to select the needed items or all rows between click  + shift click, It would save us a lot of time, 
I can use a table component to get the sort and other features but still even that does not have a quick and fast multiselect option. 

### What does the proposed API look like?
dragSelect={true}
 or 
shiftSelect={true}

This simple prop would allow us to make multi selections simpler in both transfer component or Table transfer.

**Comment by anuragnandi** [comment]
This can be fixed just by switching onSelect(.....) with
 
`   onChange(selectedKeys, selectedRows) {
            onItemSelectAll(selectedKeys, selectedRows)
 },
`

**Comment by tompenzer** [comment]
I don't know what @anuragnandi meant with the last comment, but I don't understand how that solution is supposed to work. It seems like an oversight that this feature is not officially supported.
